client/gutenberg/editor/controller.js

```diff
@@ -325,3 +326,29 @@ export const redirectSiteEditor = async ( context, next ) => {
 	const siteEditorUrl = addQueryArgs( queryArgs, `${ siteAdminUrl }site-editor.php` );
 	return location.assign( siteEditorUrl );
 };
+/**
+ * Redirect the logged user to the permalink of the post, page, custom post type if the post is published.
+ *
+ * @param {Object} context Shared context in the route.
+ * @param {Function} next  Next registered callback for the route.
+ * @returns {*}            Whatever the next callback returns.
```

Comment:
This is only the case if the redirect conditions aren't fulfilled. If we end up redirecting, it will return `undefined`.